src/datasets/io/handler.py

```diff
@@ -0,0 +1,19 @@
+import gzip
+
+
+class IOHandler:
```

Comment:
Just a naming suggestion, feel free to ignore if `IOHandler` sounds better to you
```suggestion
class open_compressed:
```

Reply:
I also think this could simply be a function that returns the output of `open` or `gzip.open`, what do you think ?

Reply:
Not sure about this but I think we'll need to close file too after writing which we aren't doing in `json.py` hence `__exit__` thing in context manager

Reply:
`IOHandler` is not just opening compressed files but uncompressed as well :p